### DISCUSSION TITLE
Subcommands could use an overhaul

### DISCUSSION THREAD

**Discussion by wycats** [erikhuda/thor#843] [Ideas]
Currently, subcommands are mostly tacked on and could use some proper thinking behind them.

**Comment by indirect** [comment]
That would be me :P IMO, the entire #invoke structure needs to be reworked in a way that makes thor commands modular and extensible. At that point, implementing subcommands in a non-crazy way should be relatively trivial. Heh.

**Comment by carllerche** [comment]
They don't work with Thor::Group at all

**Comment by aiwilliams** [comment]
So, I've wasted a good deal of my time today trying to make subcommands, with useful help output, work with v0.14.6. Is there any hope for the near future? I'd love to say I could dig in and fix something, but I've only been with the thing for a few hours now.
